### DISCUSSION TITLE
Email Templates for Backend in Django Admin?

### DISCUSSION THREAD

**Discussion by BethanyG** [codebuddies/backend#181] [Ideas]
Came across this library when I was drumming up links for issue #180.  It (apparently) adds the ability to create and administer email templates from the Django Admin interface.  It also has a very basic "mass mailing" functionality that could be used for quick "update" type emails or "newsletter" type emails.  Don't know if we really need it, but in thinking about the various emails we (might) need to create for the actions users take in the system, it might make our lives easier.  

This library **does not** remove the requirement to learn the [Django Templating Language](https://docs.djangoproject.com/en/3.0/ref/templates/language/), but it does remove the overhead of having to find the email templates and craft them separately.

[django-emailtemplates](https://pypi.org/project/django-emailtemplates/)

That being said, having the emails as a technical task that is "beginner friendly" might also be a good strategy for onboarding newer users to Django, since they will have to navigate the file tree, extract meaningful variables, and make commits for the new files.  **Thoughts?**

Anticipating the following emails:

- New User Registration Email Validation
- User Password Reset Request
- User Forgotten Password Request
- Welcome New User Email
- Hangout Scheduled Confirmation
- Another user RSVPd to Hangout (Hangout owner notified)
- Confirmation of an RSVP to a Hangout (user RSVP confirmation)
- Notification a Hangout has been rescheduled or cancelled
- New Hangout Scheduled Alert (in the scenario where someone is following a topic)
- Discussions Summary (??  thinking about how we keep members in the loop ??)
- "Newsletter" (?? again, thinking about how we keep members in the loop ??)
- Pairing request (??? would we do this ??)
- Communication from another member (?? would we do this from the interface or no ???)

**Comment by lpatmo** [comment]
> That being said, having the emails as a technical task that is "beginner friendly" might also be a good strategy for onboarding newer users to Django, since they will have to navigate the file tree, extract meaningful variables, and make commits for the new files.

Agree with this. 

I think if editing each of the files for these transactional emails gets really cumbersome, we could explore adding this so we can manage in /admin as superusers, but until then, it's a good beginner issue.

Note on the newsletter -- I lean towards managing it via a "marketing email" newsletter manager rather than via transactional emails using the same "from" address, since they're more likely to be marked as spam and have lower open rates.

**Comment by BethanyG** [comment]
Adding some additional links here.  The topic of Responsive email templates was brought up, so adding a few references here around that:

-  [Templated Email Plugin for Django](https://github.com/vintasoftware/django-templated-email) & [Related Article](https://bradwhittington.wordpress.com/2011/05/01/handling-email-templates-and-transactional-mail-in-django/)
-  [Zurb Foundation Responsive Email Templates for Django](https://pypi.org/project/django-zurb-foundation/)
-  [Dev.to Article on Transactional Emails in Django](https://dev.to/vinaypai/django-transactional-emails-made-easy-1pf1)

**Comment by BethanyG** [comment]
One thought while I have it in my head:  one thing I really like from exercism is their "weekly mentoring update" emails that track my activity on the site.  Whether they be from Django or from a separate program, something like that for users of codebuddies (how many hangouts made/attended, how many resources, how many learnings/notes, how many pairings) might be a fun thing to consider.  For examples sake, let me paste one here:

![exercism update](https://user-images.githubusercontent.com/5923094/92673224-78f9a480-f2cf-11ea-83e8-070b01c1fd18.png)

**Reply by lpatmo** [reply]
That's awesome! Having this (transactional) email also forces us to be explicit about the key metrics for CBV3 website usage -- and it may very well be # of hangouts in the beginning, followed by other things we'll consider important as we build out the rest of the app 🤔

**Reply by BethanyG** [reply]
Yeah - agree!  And we can consider extending/adding to it with more "personal goals" focused content -- like things attached to learning paths or resources, etc.
